src/basilisp/cli.py

```diff
@@ -747,14 +751,29 @@ def test(
 @_subcommand(
     "test",
     help="run tests in a Basilisp project",
-    description="Run tests in a Basilisp project.",
+    description=textwrap.dedent(
+        """Run tests in a Basilisp project.
+
+        Any options not recognized by Basilisp and all positional arguments will
+        be collected and passed on to PyTest. It is possible to directly signal
+        the end of option processing using an explicit `--` as in:
+
+            `basilisp test -p other_dir -- -k vector`
+
+        This can be useful to also directly execute PyTest commands with Basilisp.
+        For instance, you can directly print the PyTest command-line help text using:
+
+            `basilisp test -- -h`
+
+        If all options are unambiguous (e.g. they are only either used by Basilisp
+        or by PyTest), then you can omit the `--`:
+
+            `basilisp test -k vector -p other_dir`"""
```

Comment:
This also addresses [the comment](https://github.com/basilisp-lang/basilisp/issues/1069#issuecomment-2375202293) about documenting the `--` behavior.